### ISSUE TITLE
Find a Store search box needs label other than the placeholder

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13439]
**Describe the bug**

- Automated test detects missing label.
- Placeholders shouldn't be used as labels.

**Tell us the version of Spartacus**
- Library version: 4.0.0

**Expected behavior**

- Associate a label, or use aria-label.
- Use aria-describedby if more instructions are needed. (e.g. enter a promo code here if you have one)
- Screen Reader vocalizes field and required action.

**Screenshots**
![Screen Shot 2021-08-05 at 6 33 01 PM](https://user-images.githubusercontent.com/15113219/128570520-64138a97-99f3-4325-a5e1-127b29fba930.png)

**Comment by saad-mo** [comment]
Solution: It's easier to add aria-label without disrupting the existing UI.  add `aria-label to search input in `store-finder-search.component.html`

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.
![FindStore](https://user-images.githubusercontent.com/45409515/144891152-eb164281-cf11-4b4f-9a09-dc2501d4b79e.png)
